servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpHeaders.java

```diff
@@ -602,4 +604,24 @@ default HttpHeaders addSetCookie(final CharSequence name, final CharSequence val
      * @return {@code true} if at least one entry has been removed.
      */
     boolean removeSetCookies(CharSequence name, CharSequence domain, CharSequence path);
+
+    /**
+     * Returns the total length in bytes of all the header names and values combined.
+     * <p>
+     * Note that this does not include whitespace like new lines or spaces between name and value.
+     *
+     * @return the total length of headers in bytes.
+     */
+    default int totalLength() {
+        if (isEmpty()) {
+            return 0;
+        }
+
+        int length = 0;
+        for (final Entry<CharSequence, CharSequence> entry : this) {
+            length += entry.getKey().length();
+            length += entry.getValue().length();
+        }
+        return length;
+    }
```

Comment:
It's good enough for default implementation, but the main motivation for this proposal was that 2 main implementations we have today (`DefaultHttpHeaders` and `NettyH2HeadersToHttpHeaders`) should keep track of the total length on every modification instead of calculating it at runtime. It's possible to do for `DefaultHttpHeaders`, but for `NettyH2HeadersToHttpHeaders` it's tricky for `set` cases bcz we don't know if `set` overrides the previous value or not. It may require a change in Netty first.

Until that, there is not much value in adding new API if we can not gain more performance. For now, it can stay as internal utility.

Reply:
Incrementing/decrementing an integer is quite trivial is done on every mutation, it's like updating `size` in every `Collection`. If we end up needing this for every request, it will worth it.